Corona-Warn-App/src/main/java/de/rki/coronawarnapp/transaction/RiskLevelTransaction.kt

```diff
@@ -374,38 +387,59 @@ object RiskLevelTransaction : Transaction() {
 
-            // get the high risk score class
-            val highRiskScoreClass =
-                riskScoreClassification.riskClassesList.find { it.label == "HIGH" }
-                    ?: throw RiskLevelCalculationException(IllegalStateException("no high risk score class found"))
-
-            // if the calculated risk score is above the defined level threshold we return the high level risk score
-            if (riskScore >= highRiskScoreClass.min && riskScore <= highRiskScoreClass.max) {
-                Timber.v("$riskScore is above the defined min value ${highRiskScoreClass.min}")
-                return@executeState INCREASED_RISK
-            } else if (riskScore > highRiskScoreClass.max) {
-                throw RiskLevelCalculationException(
-                    IllegalStateException("risk score is above the max threshold for score class")
-                )
-            }
+            return@executeState getRiskLevel(
+                riskLevelCalculation,
+                riskScoreAnalysis,
+                attenuationParameters,
+                exposureSummary,
+                riskScoreClassification
+            )
+        }
 
-            Timber.v("$transactionId - INCREASED_RISK not applicable")
-            return@executeState UNDETERMINED
+    public fun getRiskLevel(
+        riskLevelCalculation: RiskLevelCalculation,
+        riskScoreAnalysis: RiskScoreAnalysis,
+        attenuationParameters: ApplicationConfigurationOuterClass.AttenuationDuration,
+        exposureSummary: ExposureSummary,
+        riskScoreClassification: ApplicationConfigurationOuterClass.RiskScoreClassification
+    ): RiskLevel {
+        // calculate the risk score based on the values collected by the Google EN API and
+        // the backend configuration
+        val riskScore = riskLevelCalculation.calculateRiskScore(
+            attenuationParameters,
+            exposureSummary
+        ).also {
+            Timber.v(TAG, "calculated risk with the given config: $it")
         }
 
+        // get the high risk score class
+        val highRiskScoreClass =
+            riskScoreClassification.riskClassesList.find { it.label == "HIGH" }
+                ?: throw RiskLevelCalculationException(IllegalStateException("no high risk score class found"))
+
+        // if the calculated risk score is above the defined level threshold we return the high level risk score
+        if (riskScoreAnalysis.withinDefinedLevelThreshold(
+                riskScore,
+                highRiskScoreClass.min,
+                highRiskScoreClass.max
+            )
+        ) {
+            Timber.v("$riskScore is above the defined min value ${highRiskScoreClass.min}")
+            return INCREASED_RISK
+        } else if (riskScore > highRiskScoreClass.max) {
+            throw RiskLevelCalculationException(
+                IllegalStateException("risk score is above the max threshold for score class")
+            )
+        }
```

Comment:
What do you think about moving the whole threshold check to `RiskScoreAnalysis`, i.e.

```kotlin
RiskScoreAnalysis.isIncreasedRisk(
    riskScore: Double,
    riskScoreClassification: ApplicationConfigurationOuterClass.RiskScoreClassification,
) : Boolean

```

Reply:
good idea, but not for this feature